arduino/scout/scout.ino

```diff
@@ -67,26 +67,36 @@ void updateFromAnalogInputs() {
 }
 
 void loop() {
+  uint8_t i;
+  uint8_t size;
+
   buffer.populate();
+  size = buffer.getSize();
 
   // TODO: do this less often...
   updateFromAnalogInputs();
 
   if (printToSerial) {
-    frequency.print();
+    for (i = 0; i < size; i++) {
+      frequency[i].print();
+    }
   }
 
   if (buffer.isEmpty()) {
-    if (!glideOnFreshKeyPresses) {
-      frequency.reset();
-    }
-
-    noTone(SPEAKER_PIN);
     digitalWrite(PLAYING_INDICATOR_LED, LOW);
   } else {
-    frequency.update(notes.get(buffer.getFirst()) / 4 * pow(2, octave), glide);
-
-    tone(SPEAKER_PIN, frequency.get());
     digitalWrite(PLAYING_INDICATOR_LED, HIGH);
   }
+
+  for (i = 0; i < size; i++) {
+    frequency[i].update(notes.get(buffer.getElement(i)) / 4 * pow(2, octave), glide);
+    speaker_preload[i] = frequency[i].getPeriod();
```

Comment:
Setting speaker_preload[i] is a substitute for the call to tone for each of the oscillating pins. I guess I could hide this behind a function call?